mibidata/mibi_image.py

```diff
@@ -239,30 +237,35 @@ def channel_inds(self, channels):
             KeyError: Raised if channels are not all found in image.
         """
         try:
-            return self._index[channels]
-        except (KeyError, TypeError):
+            return self._channels.index(channels)
+        except ValueError:
             pass
         try:
-            return self._target_index[channels]
-        except (KeyError, TypeError, AttributeError):
+            return self.targets.index(channels)
+        except (ValueError, AttributeError):
             pass
         try:
-            return self._mass_index[channels]
-        except (KeyError, TypeError, AttributeError):
+            return self.masses.index(channels)
+        except (ValueError, AttributeError):
             pass
         try:
-            return [self._index[i] for i in channels]
-        except KeyError:
+            return [self._channels.index(i) for i in channels]
+        except ValueError:
             pass
         try:
-            return [self._mass_index[i] for i in channels]
-        except (KeyError, AttributeError):
+            return [self.masses.index(i) for i in channels]
+        except (ValueError, AttributeError):
             pass
         try:
-            return [self._target_index[i] for i in channels]
-        except (KeyError, AttributeError):
-            raise KeyError(f'Subset of channels, targets or massses not found '
-                           f'matching {channels}')
+            return [self.targets.index(i) for i in channels]
+        except (ValueError, AttributeError):
+            error_msg = f'Subset of channels, targets or massses not found ' \
+                        f'matching {channels}.'
+            if self.targets is None:
+                error_msg += ' Note: channels were indexed with target names ' \
+                             'only.'
+
+            raise KeyError(error_msg)
```

Comment:
I think it'll be clearer if we switch based on whether targets are present rather than appending. Something like:
```
if self.targets is None:
    error_msg = f'Channels were indexed with targets only (no masses were given), available targets are {channels}'
else:
    error_msg = f'Subset of channels, targets or massses not found matching {channels}/'

Reply:
Will do